webapp/components/wpt-runs.html

```diff
@@ -218,58 +232,61 @@
       async ready() {
         super.ready();
         this.loadData();
+        this._createMethodObserver('testRunsLoaded(testRuns, testRuns.*)');
       }
 
       async loadData() {
-        return this.load(this.loadRuns().then(testRuns => {
-          let browsers = new Set();
-          // Group the runs by their revision/SHA
-          let testRunsBySha = testRuns.reduce((accum, results) => {
-            browsers.add(results.browser_name);
-            if (!accum[results.revision]) {
-              accum[results.revision] = {};
-            }
-            if (!accum[results.revision][results.browser_name]) {
-              accum[results.revision][results.browser_name] = [];
-            }
-            accum[results.revision][results.browser_name].push(results);
-            return accum;
-          }, {});
+        return this.load(this.loadRuns());
+      }
 
-          // We flatten into an array of objects so Polymer can deal with them.
-          const firstRunDate = runs => {
-            return Object.values(runs)
-              .reduce((oldest, runs) => {
-                for (const time of runs.map(r => new Date(r.time_start))) {
-                  if (time < oldest) {
-                    oldest = time;
-                  }
-                }
-                return oldest;
-              }, new Date()); // Existing runs should be historical...
-          };
-          const flattened = Object.entries(testRunsBySha)
-            .map(([sha, runs]) => ({ sha, runs, firstRunDate: firstRunDate(runs) }))
-            .sort((a, b) => b.firstRunDate.getTime() - a.firstRunDate.getTime());
+      testRunsLoaded(testRuns) {
+        let browsers = new Set();
+        // Group the runs by their revision/SHA
+        let testRunsBySha = testRuns.reduce((accum, results) => {
```

Comment:
Now that `this.testRuns` is renamed to `this.testRunsBySha`, we should rename this local variable to avoid confusion as they are not the same (`this.testRunsBySha` is the flattened `testRunsBySha`).

Reply:
Renamed.